flake.nix

```diff
@@ -279,7 +286,7 @@
                     ${server}/bin/ctl-server \
                       --port ${toString ctlServer.port} \
                       --node-socket ${nodeSocketPath} \
-                      --network-id 1097911063
+                      --network-id ${toString config.network.magic}
```

Comment:
I don't think this will work for mainnet. You need to check if the `magic` attribute is set or if it's `null`. Also we need to add something to the documentation indicating that `mainnet` should not have `magic` set

Reply:
To elaborate: the correct flag for ctl-server is `--network-id mainnet`